experimental/displaynames/src/displaynames.rs

```diff
@@ -75,20 +73,14 @@ impl RegionDisplayNames {
         ]
     );
 
-    /// Returns the locale specific display name of a region for a given string.
-    /// This function is locale-sensitive.
-    pub fn of(&self, code: &str) -> Option<&str> {
-        match <TinyAsciiStr<3>>::from_str(code) {
-            Ok(key) => {
-                let data = self.region_data.get();
-                match self.options.style {
-                    Some(Style::Short) => data.short_names.get(&key),
-                    _ => None,
-                }
-                .or_else(|| data.names.get(&key))
-            }
-            Err(_) => None,
+    /// Returns the display name of a region.
+    pub fn of(&self, region: Region) -> Option<&str> {
+        let data = self.region_data.get();
+        match self.options.style {
+            Some(Style::Short) => data.short_names.get(&region.into()),
+            _ => None,
         }
+        .or_else(|| data.names.get(&region.into()))
```

Comment:
Because we are removing the key-value pairs with the same string, what happens if the user calls this API for one of that keys? This would just return an empty string?

Reply:
This would return `None` and then the user will have to figure out their own fallback. Ideally this should implement the full algorithm specified in https://www.unicode.org/reports/tr35/tr35-general.html#Display_Name_Elements and return `Cow<str>` unconditionally.